module-network.tf

```diff
@@ -0,0 +1,187 @@
+# Copyright (c) 2017, 2023 Oracle Corporation and/or its affiliates.
+# Licensed under the Universal Permissive License v 1.0 as shown at https://oss.oracle.com/licenses/upl
+
+data "oci_core_vcns" "all" {
+  count          = var.create_vcn ? 0 : 1
+  compartment_id = local.compartment_id
+  display_name   = var.vcn_name
+  state          = "AVAILABLE"
+}
+
+data "oci_core_vcn" "oke" {
+  count  = var.create_vcn ? 0 : 1
+  vcn_id = var.vcn_id
+}
+
+data "oci_core_nat_gateways" "all" {
+  compartment_id = local.compartment_id
+  display_name   = var.nat_gateway_display_name
+  state          = "AVAILABLE"
+  vcn_id         = local.vcn_id
+}
+
+data "oci_core_internet_gateways" "all" {
+  compartment_id = local.compartment_id
+  display_name   = var.internet_gateway_display_name
+  state          = "AVAILABLE"
+  vcn_id         = local.vcn_id
+}
+
+data "oci_core_route_tables" "all" {
+  compartment_id = local.compartment_id
+  state          = "AVAILABLE"
+  vcn_id         = local.vcn_id
+}
+
+locals {
+  vcn_id = (var.create_vcn ? one(module.vcn[*].vcn_id)
+    : coalesce(var.vcn_id, "none") != "none" ? var.vcn_id
+    : one(one(data.oci_core_vcns.all[*].virtual_networks)[*].id)
```

Comment:
this is assuming that a vcn with the same display name will be the one you're looking for. However, as mentioned above, `display_name` is not unique.

Reply:
Yes, we should fail it. A vcn_id is something specific, it means users know what they are doing whereas looking by name, there's a potential to create something in vcn where it was not intended to be. More importantly, when you look up by name, sort order is not guaranteed and even if you sort it alphanumerically, it may not end up being the vcn the user intended the cluster to go to.

Reply:
Done.